common/src/main/java/io/netty/util/ResourceLeakDetectorFactory.java

```diff
@@ -57,30 +57,47 @@ public static void setResourceLeakDetectorFactory(ResourceLeakDetectorFactory fa
     /**
      * Returns a new instance of a {@link ResourceLeakDetector} with the given resource class.
      *
-     * @param resource - the resource class used to initialize the {@link ResourceLeakDetector}
-     * @param <T> - the type of the resource class
-     * @return - a new instance of {@link ResourceLeakDetector}
+     * @param resource the resource class used to initialize the {@link ResourceLeakDetector}
+     * @param <T> the type of the resource class
+     * @return a new instance of {@link ResourceLeakDetector}
      */
     public final <T> ResourceLeakDetector<T> newResourceLeakDetector(Class<T> resource) {
-        return newResourceLeakDetector(resource, ResourceLeakDetector.DEFAULT_SAMPLING_INTERVAL, Long.MAX_VALUE);
+        return newResourceLeakDetector(resource, ResourceLeakDetector.DEFAULT_SAMPLING_INTERVAL);
     }
 
     /**
+     * @deprecated Use {@link #newResourceLeakDetector(Class, int)} instead.
+     * <p>
      * Returns a new instance of a {@link ResourceLeakDetector} with the given resource class.
      *
-     * @param resource - the resource class used to initialize the {@link ResourceLeakDetector}
-     * @param samplingInterval - the interval on which sampling takes place
-     * @param maxActive - the maximum active instances
-     * @param <T> - the type of the resource class
-     * @return - a new instance of {@link ResourceLeakDetector}
+     * @param resource the resource class used to initialize the {@link ResourceLeakDetector}
+     * @param samplingInterval the interval on which sampling takes place
+     * @param maxActive This is deprecated and will be ignored.
+     * @param <T> the type of the resource class
+     * @return a new instance of {@link ResourceLeakDetector}
      */
+    @Deprecated
     public abstract <T> ResourceLeakDetector<T> newResourceLeakDetector(
             Class<T> resource, int samplingInterval, long maxActive);
 
     /**
+     * Returns a new instance of a {@link ResourceLeakDetector} with the given resource class.
+     *
+     * @param resource the resource class used to initialize the {@link ResourceLeakDetector}
+     * @param samplingInterval the interval on which sampling takes place
+     * @param <T> the type of the resource class
+     * @return a new instance of {@link ResourceLeakDetector}
+     */
+    @SuppressWarnings("deprecation")
+    public <T> ResourceLeakDetector<T> newResourceLeakDetector(Class<T> resource, int samplingInterval) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused method parameter "samplingInterval". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1172)